src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -3709,4 +3752,18 @@ private boolean unicodeConversionRequired(int jdbcType, SSType ssType) {
                 || java.sql.Types.LONGNVARCHAR == jdbcType)
                 && (SSType.NCHAR == ssType || SSType.NVARCHAR == ssType || SSType.NVARCHARMAX == ssType));
     }
+
+    private static String getHashedSecret(String[] secrets) throws SQLServerException {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            for (String secret : secrets) {
+                if (null != secret) {
+                    md.update(secret.getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+                }
+            }
+            return new String(md.digest());
+        } catch (NoSuchAlgorithmException e) {
+            throw new SQLServerException(SQLServerException.getErrString("R_NoSHA256Algorithm"), e);
+        }
+    }
```

Comment:
this is a dupe of the same method in SQLServerMSAL4JUtils.java can we refactor that into a common place